### DISCUSSION TITLE
Is it better than open ai whisper model

### DISCUSSION THREAD

**Discussion by Greatz08** [speechbrain/speechbrain#2448] [Q&A]
i have tested whisper model in small and medium and it did show great speech recognition capabillity so is this solution better or has different usecase? Can ANyone explain in easiest and detailed way possible.

**Comment by Adel-Moumen** [comment]
> i have tested whisper model in small and medium and it did show great speech recognition capabillity so is this solution better or has different usecase? Can ANyone explain in easiest and detailed way possible.

Hi, I am also experiencing the same issue... I will need to explore more what is going on... but I don't have the time right now. Will keep you updated!

**Reply by Greatz08** [reply]
@Adel-Moumen  i will be waiting for your response.Take your time to test and then respond back :-)

**Reply by Adel-Moumen** [reply]
Hello @HakaishinShwet, I found some issues in the decoding process. Some tokens were not removed during decoding... I made a WIP PR https://github.com/speechbrain/speechbrain/pull/2450 ; this PR is also adding a bunch of things (e.g. detect language with Whisper) and making the code a lot faster! Support of flash attention, KV caching, etc. However, I need to spend more time on it. I plan to refactor a bit the code & I'd like to add new things as well (e.g. prompting/prefix in decoding) and support of these new features with beam search. I will also spend time on the fine tuning script. I need to double check that everything is correct. Finally, I will also modify our WhisperASR inference interface so that you can perform long form ASR (chunking etc). I'll keep you updated! :)

**Reply by Adel-Moumen** [reply]
Hello @HakaishinShwet, #2450 has been merged. Feel free to fetch the latest SB version in order to use the reworked whisper implementation. Let me know if you need some help. I got some very good results on commonvoice french / italian and librispeech.
